### PR TITLE
Add camel-kafka integration tests

### LINKED ISSUE #347: Add camel-kafka integration tests
camel-kafka integration tests

### PR DESCRIPTION
**Pull request by f2par0**
Fixes #347 
Creates a producer / consumer integration test for kafka

### CODE REVIEW

File: tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                        .withEnv("KAFKA_LISTENER_SECURITY_PROTOCOL_MAP", "CONTROLLER:PLAINTEXT,INTERNAL:PLAINTEXT,EXTERNAL_SAME_HOST:PLAINTEXT")
+                        .withEnv("KAFKA_INTER_BROKER_LISTENER_NAME", "INTERNAL")
+                        .withEnv("KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR", "1")
+                        // in case more logs needed
+                        //.withEnv("KAFKA_LOG4J_LOGGERS","org.apache.kafka=DEBUG,kafka.request.logger=DEBUG,kafka.controller=DEBUG, kafka.coordinator=DEBUG,kafka.log=DEBUG,kafka.server=DEBUG, state.change.logger=DEBUG")
+                        ;
+            kafkaContainer.setWaitStrategy(
+                    new LogMessageWaitStrategy()
+                            .withRegEx(".*Kafka Server started.*")
+                            .withStartupTimeout(Duration.ofSeconds(10)));
```
Comment (essobedo) [inline tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java:79]: This sounds overcomplicated, why not simply use the dedicated test container for it called `KafkaContainer` with a confluent image like in this blog post https://testcontainers.com/guides/testing-spring-boot-kafka-listener-using-testcontainers/#_write_test_for_kafka_listener

File: tests/features/camel-kafka/src/main/java/org/apache/karaf/camel/test/CamelKafkaRouteSupplier.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        return builder ->
+                builder.fromF("kafka://%s?autoOffsetReset=earliest", KAFKA_TOPIC)
+                        .log("received kafka message ${body}");
+    }
+
+    @Override
+    protected void configureProducer(RouteBuilder builder, RouteDefinition producerRoute) {
+        producerRoute.log("calling kafka topic")
+                .setBody(builder.constant("OK"))
+                .toF("kafka://%s?", KAFKA_TOPIC)
```
Comment (essobedo) [inline tests/features/camel-kafka/src/main/java/org/apache/karaf/camel/test/CamelKafkaRouteSupplier.java:60]: Do we need the trailing question mark?

File: tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java (hunk 1 above)
Comment (f2par0) [reply tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java]: yes, it's much easier, I used the official image instead

File: tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    @Test
+    public void testResultMock() throws Exception {
+        assertMockEndpointsSatisfied();
+    }
+
+    public static final class ExternalResourceProviders {
+
+        public static GenericContainerResource createKafkaContainer() {
+            final KafkaContainer kafkaContainer =
+                    new KafkaContainer( DockerImageName.parse("confluentinc/cp-kafka:7.6.1"));
```
Comment (essobedo) [inline tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java:55]:
```suggestion
                    new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.6.1"));
```

File: tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import org.testcontainers.utility.DockerImageName;
+
+@CamelKarafTestHint(externalResourceProvider = CamelKafkaITest.ExternalResourceProviders.class)
+@RunWith(PaxExamWithExternalResource.class)
+@ExamReactorStrategy(PerClass.class)
+public class CamelKafkaITest extends AbstractCamelSingleFeatureResultMockBasedRouteITest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CamelKafkaITest.class);
+
+    private static final int KAFKA_PORT = 9092;
```
Comment (essobedo) [inline tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java:39]:
```suggestion
```

File: tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+
+@CamelKarafTestHint(externalResourceProvider = CamelKafkaITest.ExternalResourceProviders.class)
+@RunWith(PaxExamWithExternalResource.class)
+@ExamReactorStrategy(PerClass.class)
+public class CamelKafkaITest extends AbstractCamelSingleFeatureResultMockBasedRouteITest {
+
+
```
Comment (essobedo) [inline tests/features/camel-kafka/src/test/java/org/apache/karaf/camel/itest/CamelKafkaITest.java:37]:
```suggestion
```

**Review by essobedo — APPROVED** [review]
Definitively better 👍

**Pull request merged**